### ISSUE TITLE
test4 is not checking correctness

### ISSUE THREAD

**Issue by iezepov** [rust-lang/rustlings#247]
Hi!

It's possible to "pass" test4 with incorrect macro. For example this gives message that the code is successfully compile.

```
macro_rules! my_macro {                                                                                                                 
    ($e:expr) => {
        $e
    }
}

fn main() {
    if my_macro!("world!") != "Hello world!" {
        panic!("Oh no! Wrong output!");
    }
 }
```

It compiles, yes, but it's obviously incorrect

**Comment by fmoko** [comment]
I don't see this as a pressing issue. Obviously you can write a very simply macro to pass the test in a pass-through fashion, but the point of Rustlings is to experiment with different solutions to exercises.

Still, if you want to take a shot at improving this exercise, I'd be happy to take a look at it!

**Comment by Fat-Zer** [comment]
@fmoko, IMHO it would be nice to replace the main function with a test. It's actually what I have done straight away when took the exercise...

**Comment by olamiko** [comment]
@fmoko, I had a look at this exercise and tried to improve it. I noticed that the example @iezepov gave only compiles and runs successfully in the rustlings program. It fails locally as a standalone program. Do you know why this happens?

**Comment by jrvidal** [comment]
This was fixed in #276
